### PR TITLE
Connecting app to be

### PR DESCRIPTION
**Pull request by tmaciulis22**
All functionality is connected to BE, except for doctor's app.

One bug is still left unfixed. You cannot for some reason sign in with credentials, except when you register a user through Postman. Andrius will try to fix it.

Besides, I made regex a bit easier and some additional UI changes

NOTE: Update your `secrets.json` file, when you pulled this branch.

### CODE REVIEW

File: API/MoodisAPI/apiMoodis/apiMoodis/Controllers/GroupController.cs [hunk 1]
```diff
@@ -32,8 +36,42 @@ public IHttpActionResult GetByIdGroup(string id)
         {
             using (DatabaseContext dbContext = new DatabaseContext())
             {
-                var entity = dbContext.Groups.Single(group => group.Id == id);
-                return Ok(entity);
+                try
+                {
+                    var entity = dbContext.Groups.Single(g => g.Id == id);
```
Comment (Pilkas-git) [inline API/MoodisAPI/apiMoodis/apiMoodis/Controllers/GroupController.cs]: Change naming of g to group.

File: API/MoodisAPI/apiMoodis/apiMoodis/Controllers/GroupController.cs [hunk 2]
```diff
@@ -32,8 +36,42 @@ public IHttpActionResult GetByIdGroup(string id)
...
+        }
+
+        [HttpGet]
+        public IHttpActionResult GetDoctorGroups(string doctorId)
+        {
+            using (DatabaseContext dbContext = new DatabaseContext())
+            {
+                try
+                {
+                    var groups = dbContext.Groups.TakeWhile(e => e.DoctorId == doctorId).Select(group => new GroupFE()
```
Comment (Pilkas-git) [inline API/MoodisAPI/apiMoodis/apiMoodis/Controllers/GroupController.cs]: sane here

File: API/MoodisAPI/apiMoodis/apiMoodis/Controllers/UserController.cs [hunk 3]
```diff
@@ -1,76 +1,238 @@
...
         [HttpGet]
         public IHttpActionResult GetByIdUser(string id)
         {
             using (DatabaseContext dbContext = new DatabaseContext())
             {
-                var entity = dbContext.Users.Include(item => item.ImageInfos).Single(user => user.Id == id);
-                return Ok(entity);
+                try
+                {
+                    var entity = dbContext.Users.Include(item => item.ImageInfos).Single(u => u.Id == id);
```
Comment (Pilkas-git) [inline API/MoodisAPI/apiMoodis/apiMoodis/Controllers/UserController.cs]: naming.

File: API/MoodisAPI/apiMoodis/apiMoodis/Controllers/UserController.cs [hunk 4]
```diff
@@ -1,76 +1,238 @@
...
+        [HttpGet]
+        public IHttpActionResult GetUserByPersonId(string personId)
         {
             using (DatabaseContext dbContext = new DatabaseContext())
             {
-                var entity = dbContext.Users.Include(item => item.ImageInfos).Single(user => user.Username == username);
-                return Ok(entity);
+                try
+                {
+                    var entity = dbContext.Users.Include(item => item.ImageInfos).Single(u => u.PersonId == personId);
```
Comment (Pilkas-git) [inline API/MoodisAPI/apiMoodis/apiMoodis/Controllers/UserController.cs]: naming

File: API/MoodisAPI/apiMoodis/apiMoodis/Controllers/ImageInfoController.cs [hunk 5]
```diff
@@ -45,28 +95,70 @@ public IHttpActionResult PostImageInfo([FromBody] ImageInfo imageInfo)
...
             {
-                var entity = dbContext.ImageInfos.Single(e => e.Id == id);
-                entity.DateAsString = imageInfo.DateAsString;
-                entity.UserId = imageInfo.UserId;
-                entity.ImagePath = imageInfo.ImagePath;
-                dbContext.SaveChanges();
-                return Ok(entity);
+                try
+                {
+                    var entity = dbContext.ImageInfos.Single(e => e.Id == id);
```
Comment (Pilkas-git) [inline API/MoodisAPI/apiMoodis/apiMoodis/Controllers/ImageInfoController.cs]: naming.

File: API/MoodisAPI/apiMoodis/apiMoodis/Controllers/ImageInfoController.cs [hunk 6]
```diff
@@ -75,10 +167,21 @@ public IHttpActionResult DeleteImageInfo(string id)
         {
             using (DatabaseContext dbContext = new DatabaseContext())
             {
-                var entity = dbContext.Groups.Single(e => e.Id == id);
-                dbContext.Groups.Remove(entity);
-                dbContext.SaveChanges();
-                return Ok();
+                try
+                {
+                    var entity = dbContext.Groups.Single(e => e.Id == id);
```
Comment (Pilkas-git) [inline API/MoodisAPI/apiMoodis/apiMoodis/Controllers/ImageInfoController.cs]: naming

File: API/MoodisAPI/apiMoodis/apiMoodis/Controllers/UserController.cs [hunk 7]
```diff
@@ -1,76 +1,238 @@
...
         [HttpPost]
-        public IHttpActionResult PostUser([FromBody] User user)
+        [Route("api/user/register")]
+        public IHttpActionResult PostUser([FromBody] RegisterRequest request)
         {
             try
             {
                 using (DatabaseContext dbContext = new DatabaseContext())
                 {
+                    var isAlreadyRegistered = dbContext.Users.Any(u => u.Username == request.Username);
```
Comment (Pilkas-git) [inline API/MoodisAPI/apiMoodis/apiMoodis/Controllers/UserController.cs]: naming

File: API/MoodisAPI/apiMoodis/apiMoodis/Controllers/UserController.cs [hunk 8]
```diff
@@ -1,76 +1,238 @@
...
-                entity.Username = user.Username;
-                entity.Password = user.Password;
-                entity.GroupName = user.GroupName;
-                entity.PersonGroupId = user.PersonGroupId;
-                dbContext.SaveChanges();
-                return Ok();
-
+                try
+                {
+                    var entity = dbContext.Users.Single(e => e.Id == user.Id);
```
Comment (Pilkas-git) [inline API/MoodisAPI/apiMoodis/apiMoodis/Controllers/UserController.cs]: naming

File: API/MoodisAPI/apiMoodis/apiMoodis/Controllers/UserController.cs [hunk 9]
```diff
@@ -79,10 +241,21 @@ public IHttpActionResult DeleteUser(string id)
         {
             using (DatabaseContext dbContext = new DatabaseContext())
             {
-                var entity = dbContext.Users.Single(e => e.Id == id);
-                dbContext.Users.Remove(entity);
-                dbContext.SaveChanges();
-                return Ok();
+                try
+                {
+                    var entity = dbContext.Users.Single(e => e.Id == id);
```
Comment (Pilkas-git) [inline API/MoodisAPI/apiMoodis/apiMoodis/Controllers/UserController.cs]: naming

File: API/MoodisAPI/apiMoodis/apiMoodis/Controllers/GroupController.cs (hunk 1 above)
Comment (Pilkas-git) [inline API/MoodisAPI/apiMoodis/apiMoodis/Controllers/GroupController.cs]: naming.

File: API/MoodisAPI/apiMoodis/apiMoodis/Controllers/GroupController.cs (hunk 2 above)
Comment (Pilkas-git) [inline API/MoodisAPI/apiMoodis/apiMoodis/Controllers/GroupController.cs]: naming

File: API/MoodisAPI/apiMoodis/apiMoodis/Controllers/GroupController.cs [hunk 10]
```diff
@@ -44,27 +82,69 @@ public IHttpActionResult PostGroup([FromBody] Group group)
...
             using (DatabaseContext dbContext = new DatabaseContext())
             {
-                var entity = dbContext.Groups.Single(e => e.Id == id);
-                entity.GroupName = group.GroupName;
-                entity.MembersInString = group.MembersInString;
-                dbContext.SaveChanges();
-                return Ok(entity);
+                try
+                {
+                    var entity = dbContext.Groups.Single(e => e.Id == id);
```
Comment (Pilkas-git) [inline API/MoodisAPI/apiMoodis/apiMoodis/Controllers/GroupController.cs]: naming.

File: API/MoodisAPI/apiMoodis/apiMoodis/Controllers/GroupController.cs [hunk 11]
```diff
@@ -73,10 +153,21 @@ public IHttpActionResult Delete(string id)
         {
             using (DatabaseContext dbContext = new DatabaseContext())
             {
-                var entity = dbContext.Groups.Single(e => e.Id == id);
-                dbContext.Groups.Remove(entity);
-                dbContext.SaveChanges();
-                return Ok();
+                try
+                {
+                    var entity = dbContext.Groups.Single(e => e.Id == id);
```
Comment (Pilkas-git) [inline API/MoodisAPI/apiMoodis/apiMoodis/Controllers/GroupController.cs]: naming.

File: API/MoodisAPI/apiMoodis/apiMoodis/Controllers/ImageInfoController.cs [hunk 12]
```diff
@@ -33,8 +38,53 @@ public IHttpActionResult GetImageInfoById(string id)
...
+
+        [HttpGet]
+        [Route("api/imageinfo/user")]
+        public IHttpActionResult GetUserImageInfos(string userId, [FromBody] DateTime date)
+        {
+            using (DatabaseContext dbContext = new DatabaseContext())
+            {
+                try
+                {
+                    var entities = dbContext.ImageInfos.AsEnumerable().SkipWhile(e => e.UserId != userId);
```
Comment (Pilkas-git) [inline API/MoodisAPI/apiMoodis/apiMoodis/Controllers/ImageInfoController.cs]: naming.

File: API/MoodisAPI/apiMoodis/apiMoodis/Controllers/ImageInfoController.cs (hunk 5 above)
Comment (Pilkas-git) [inline API/MoodisAPI/apiMoodis/apiMoodis/Controllers/ImageInfoController.cs]: naming

File: API/MoodisAPI/apiMoodis/apiMoodis/Controllers/ImageInfoController.cs (hunk 6 above)
Comment (Pilkas-git) [inline API/MoodisAPI/apiMoodis/apiMoodis/Controllers/ImageInfoController.cs]: naming.

File: API/MoodisAPI/apiMoodis/apiMoodis/Controllers/UserController.cs (hunk 8 above)
Comment (Pilkas-git) [inline API/MoodisAPI/apiMoodis/apiMoodis/Controllers/UserController.cs]: naming.

File: API/MoodisAPI/apiMoodis/apiMoodis/Controllers/UserController.cs (hunk 9 above)
Comment (Pilkas-git) [inline API/MoodisAPI/apiMoodis/apiMoodis/Controllers/UserController.cs]: naming.

**Review by Pilkas-git — CHANGES_REQUESTED** [review]
nice work. Small naming changes left . :+1:

**Review by Pilkas-git — APPROVED** [review]
👍👍👍👍👍👍👍👍👍

**Review by AArlauskas — APPROVED** [review]
That's big. Really big. Like really fucking big. Good job, Tautvydas. Good job.

**Pull request merged**